lib/spack/llnl/url.py

```diff
@@ -457,3 +457,11 @@ def determine_url_file_extension(path: str) -> str:
 
     prefix, ext, suffix = split_url_extension(path)
     return ext
+
+
+def is_url(path_or_url: str) -> bool:
```

Comment:
I guess you're only moving things around, but imho these are such low quality helper utilities that I'd rather see them removed instead of "promoted" to llnl.*

Reply:
Made it an "implementation detail" in 7dcbf42 I agree we should revisit these functions. Can we do it in a following PR?